src/cmd/sort.rs

```diff
@@ -116,17 +116,22 @@ pub fn iter_cmp_num<'a, L, R>(mut a: L, mut b: R) -> cmp::Ordering
             (None, None) => return cmp::Ordering::Equal,
             (None, _   ) => return cmp::Ordering::Less,
             (_   , None) => return cmp::Ordering::Greater,
-            (Some(x), Some(y)) => match x.cmp(&y) {
-                cmp::Ordering::Equal => (),
-                non_eq => return non_eq,
+            (Some(x), Some(y)) => match x.partial_cmp(&y) {
+                Some(order) =>  {
+                    match order{
+                        cmp::Ordering::Equal => (),
+                        non_eq => return non_eq,
+                    }
+                },
+                None => return cmp::Ordering::Equal
```

Comment:
Was not sure if the None clause should return equal or (). This should only happen if the float can't be compared, e.g. comparing 5.0 with NaN.